cpp/src/arrow/CMakeLists.txt

```diff
@@ -49,15 +49,32 @@ install(
 #######################################
 
 ADD_ARROW_TEST(allocator-test)
+
 ADD_ARROW_TEST(array-test)
+ARROW_TEST_LINK_LIBRARIES(array-test ${BOOST_REGEX_LIBRARY})
+
 ADD_ARROW_TEST(array-decimal-test)
+ARROW_TEST_LINK_LIBRARIES(array-decimal-test ${BOOST_REGEX_LIBRARY})
+
 ADD_ARROW_TEST(buffer-test)
 ADD_ARROW_TEST(memory_pool-test)
+
 ADD_ARROW_TEST(pretty_print-test)
+ARROW_TEST_LINK_LIBRARIES(pretty_print-test ${BOOST_REGEX_LIBRARY})
+
 ADD_ARROW_TEST(status-test)
+
 ADD_ARROW_TEST(type-test)
+ARROW_TEST_LINK_LIBRARIES(type-test ${BOOST_REGEX_LIBRARY})
+
 ADD_ARROW_TEST(table-test)
+ARROW_TEST_LINK_LIBRARIES(table-test ${BOOST_REGEX_LIBRARY})
+
 ADD_ARROW_TEST(tensor-test)
+ARROW_TEST_LINK_LIBRARIES(tensor-test ${BOOST_REGEX_LIBRARY})
 
 ADD_ARROW_BENCHMARK(builder-benchmark)
+ARROW_BENCHMARK_LINK_LIBRARIES(builder-benchmark ${BOOST_REGEX_LIBRARY})
+
 ADD_ARROW_BENCHMARK(column-benchmark)
+ARROW_BENCHMARK_LINK_LIBRARIES(column-benchmark ${BOOST_REGEX_LIBRARY})
```

Comment:
You shouldn't have to link this library here, or in any library that links to libarrow. I think you only need to add boost_regex to these lists of link libs: https://github.com/apache/arrow/blob/master/cpp/CMakeLists.txt#L695